### PR TITLE
Contentful integration & offline sync

### PR DESCRIPTION
**Pull request by robbiemccorkell**
Add contentful integration and offline sync

I realised that due to the nature of the project we should try and work on any tech we realistically can without making too many assumptions on the design. One of these features will be the offline sync, which is part of our first story - event list view: https://trello.com/c/JFzDJ7GH/42-events-listing-view-all-events

- [x] Pull in and use contentful sdk to fetch event data
- [x] Add local storage of fetched event data
- Use contentful sdk sync function to fetch deltas of events
  - [x] Process new events
  - [x] Process changed events
  - [x] Process deleted events
- [x] Populate test CMS with large dataset to test offline capability
- ~~Integrate into redux logic flow?~~ Might not need redux just yet

## Pre-flight check-list

Before raising a pull request

* [x] Documentation
* [x] Unit tests
* [x] Build passing

## Pre-merge check-list

* [x] Code review (At least one :thumbsup:)
* [ ] Tester approved

### CODE REVIEW

File: scripts/generate-content.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    .getSpace(spaceId)
+    .then(async space => {
+      const { total: totalEvents } = await space.getEntries({
+        content_type: "event",
+        limit: 0
+      });
+      const toGenerate = Math.max(0, 150 - totalEvents);
+
+      console.log(`Space contains ${totalEvents} events`);
+      console.log(`Genrating ${toGenerate} events`);
```
Comment (Midnite) [inline scripts/generate-content.js]: typo

Comment (robbiemccorkell) [reply scripts/generate-content.js]: Fixed, thanks

**Comment by Midnite** [comment]
@robbiemccorkell Testing this I ran into two problems, one is that apparently by default the Contentful Management API enforces rate limits of 10 requests per second, the other is my account doesn't seem to have permission to create content (generated a personal token for content management).

**Comment by robbiemccorkell** [comment]
@Midnite Oops, hadn't realised. All developers should now be able to create content in the Pride Test contentful space.

The rate limits are expected, but the tool should wait and retry for each event it's saving. Eventually it should get past the rate limits and create all the mock events for you.

Try again and see if that works.

**Comment by Midnite** [comment]
@robbiemccorkell Thanks works wonderfully now, only have one question. Even if there are more than a 100 generated entries, the script will limit the number it deletes in one go to 100. Can't figure out if it's an intended limit or not? 

>Space contains 144 generated events
>Deleting 100 events

**Comment by robbiemccorkell** [comment]
Oh yea, I forgot to mention that. When deleting I have to fetch the IDs of all the events in the CMS, which is paginated and has a maximum page size of 100. I couldn't be bothered to automatically paginate, so right now you have to run it a couple of times to get rid of all of them!

**Comment by Midnite** [comment]
@robbiemccorkell I actually just found it in the contentful docs! :) in that case I'm happy to merge

**Comment by robbiemccorkell** [comment]
Woop

**Pull request merged**